Sources/Plasma/FeatureLib/pfPython/pyGUIControlButton.cpp

```diff
@@ -53,9 +53,9 @@ pyGUIControlButton::pyGUIControlButton(plKey objkey) : pyGUIControl(std::move(ob
 {
 }
 
-bool pyGUIControlButton::IsGUIControlButton(pyKey& gckey)
+bool pyGUIControlButton::IsGUIControlButton(const plKey& key)
 {
-    if ( gckey.getKey() && pfGUIButtonMod::ConvertNoRef(gckey.getKey()->ObjectIsLoaded()) )
+    if ( key && pfGUIButtonMod::ConvertNoRef(key->ObjectIsLoaded()) )
```

Comment:
`key` is a reference, so this can never be null, right?

Reply:
Right, but the underlying `plKeyData*` can be `nullptr`, so we must test before accessing.

Reply:
I guess behind the scenes this is actually doing `key != (plKeyData*)nullptr` and calling this operator? https://github.com/H-uru/Plasma/blob/ad3863057e51ffb65a43d30c6856d7ddb769b139/Sources/Plasma/NucleusLib/pnKeyedObject/plKey.h#L81

Reply:
It's implicitly using `operator plKeyImp*() const` to get at the underlying `fKeyData` pointer, and then does a normal bool conversion (`!= nullptr`) on that pointer.

(Coincidentally, I'm working on a change that will make this bit of `plKey` less convoluted.)

Reply:
Yes. This is a fairly common pattern in Plasma to make sure we don't crash dereferencing null keys. The only thing I did in these functions is change the argument from `plKey`'s Python wrapper class `pyKey` to `plKey` itself. We were previously and still are checking that the underlying `plKeyData` is not null.